scripts/us_census/pep/us_pep_sex/process.py

```diff
@@ -91,203 +104,325 @@ def _national_1900_1979(file_path: str) -> pd.DataFrame:
+        df = df[(df["Age"] == "All Age") &
+                (df["Year"].str.startswith("July"))].reset_index(drop=True)
+        df["Year"] = df["Year"].str.replace("July 1, ", "")
+        # dropping unwanted columns
+        df = df.drop(columns=['Age'])
         df.insert(0, 'geo_ID', 'country/USA', True)
+        float_col = df.select_dtypes(include=['float64'])
+        for col in float_col.columns.values:
+            df[col] = df[col].astype('int64')
         df['Measurement_Method'] = 'dcAggregate/CensusPEPSurvey_PartialAggregate'
-    else:
-        df = pd.read_csv(file_path,
-                         thousands=',',
-                         engine='python',
-                         skiprows=6,
-                         skipfooter=102,
-                         header=None)
+        return df
+    except Exception as e:
+        logging.fatal(f"Error processing the file {file_path}: {e}")
+
+
+def _national_2000_2010(file_path: str) -> pd.DataFrame:
+    """
+    Process and cleans the file for national 2000 to 2010.
+
+    Args:
+        file_path (str) : input file path.
+
+    Returns:
+        df (pd.DataFrame) : cleaned dataframe.
+    """
+    try:
+        df = pd.read_csv(file_path, thousands=',', skiprows=4, header=None)
         df.columns = [
-            'Age', 'Total', 'Count_Person_Male', 'Count_Person_Female',
-            'White Total', 'White Male', 'White Female', 'Black Total',
-            'Black Male', 'Black Female', 'OtherRace Total', 'OtherRace Male',
-            'OtherRace Female'
+            'SEX', 'April2000', '2000', '2001', '2002', '2003', '2004', '2005',
+            '2006', '2007', '2008', '2009', 'April2010', '2010'
         ]
+        df = df.query('SEX=="MALE" or SEX=="FEMALE"')
         # dropping unwanted columns
-        df = df.drop(columns=[
-            'Age', 'Total', 'White Total', 'White Male', 'White Female',
-            'Black Total', 'Black Male', 'Black Female', 'OtherRace Total',
-            'OtherRace Male', 'OtherRace Female'
-        ])
-        # adding geoid, year and measurement method
-        df['Year'] = year
+        df = df.drop(columns=['April2000', 'April2010', '2010'])
+        df = df.replace({
+            'SEX': {
+                'MALE': 'Count_Person_Male',
+                'FEMALE': 'Count_Person_Female'
+            }
+        })
+        # replacing rows with columns
+        # making the first row as column name
+        # to get all dataframe in one formate
+        df = df.transpose().reset_index()
+        df.columns = df.iloc[0]
+        df = df[1:]
         df.insert(0, 'geo_ID', 'country/USA', True)
+        df = df.fillna(-1)
+        float_col = df.select_dtypes(include=['float64'])
+        for col in float_col.columns.values:
+            df[col] = df[col].astype('int64')
+            df[col] = df[col].astype("str").str.replace("-1", "")
+        df.rename(columns={'SEX': 'Year'}, inplace=True)
```

Comment:
Why is the column 'SEX' being renamed to 'Year'?

Reply:
It  has been renamed accordingly to match the data frame after modification.